### PR TITLE
[test][stdlib] Add strto*_l stubs for OpenBSD.

### PR DESCRIPTION
**Pull request by 3405691582**
The template defined for Cygwin and friends does not handle failure cases
properly so the NumericParsing unit test fails.  To get the correct test
behavior, we need to use an implementation such as like in the Windows
specializations or the non-Cygwin implementation that takes a function
pointer.

However, adding additional specializations would be too wordy, and
OpenBSD doesn't have locale-dependent definitions of the relevant strto*
functions. We could add a specialization for a two-argument function
pointer, but that becomes too repetitive.

Instead, implement a few stubs and use the preprocessor, a la the
implementation for Android.

### CODE REVIEW

**Comment by compnerd** [comment]
Hmm, do you think its worth trying to unify the Android and OpenBSD paths?

**Comment by 3405691582** [comment]
I think there's a lot of scope for cleanup here, especially as well as how that `_WIN32` branch does a lot of what the generic function pointer interface does. 

Happy to take a stab at it, but it might make more sense to do it in another PR.

**Comment by compnerd** [comment]
Okay, a follow up for that would definitely be a good thing.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
⛵
